haskell/experimental/providers.bzl

```diff
@@ -0,0 +1,8 @@
+HaskellModuleInfo = provider(
+    doc = "Information about a single compiled Haskell module.",
+    fields = {
+        "object_file": "The compiled `.o` file.",
+        "interface_dir": "The import search directory for the interface file.",
```

Comment:
For a while I mistakenly thought that this was just the directory the `interface_file` is in, and this kind of confused me.

After working with it for a bit I realized that the directory structure between `interface_dir` and `interface_file` is relevant to the module import path.

I don't know if this would be obvious to most people working with `rules_haskell` or if it's worth putting some kind of explanation.

Reply:
Ah, yes, that is a misleading name. I've renamed it to `import_dir` to be consistent with the package configuration's `import-dirs` field and expanded the docstring a bit.